Fix exception not thrown by "getEntityIdentifier"

jderusse opened the pull request:
In #8201, the behavior of `getEntityIdentifier` changed to return either an `array` **OR** an instance of `EntityNotFoundException`.

But everywhere else in the project, we assume the function to return an array (here is one of the 40 occurrences I found https://github.com/doctrine/orm/blob/233d9b027692cad87c970eb660958855daeb8043/lib/Doctrine/ORM/Persisters/Entity/BasicEntityPersister.php#L1014-L1015)

This PR change the method to throws the exception instead of returning it, and change the docblock to match the description saying ` The returned value is always an array of identifier values` (https://github.com/doctrine/orm/blob/233d9b027692cad87c970eb660958855daeb8043/lib/Doctrine/ORM/UnitOfWork.php#L3053)

jderusse committed:
Fix exception not thrown by "getEntityIdentifier"

greg0ire commented:
@oojacoboo woops! Not sure how I didn't catch that during code review :sweat_smile:

greg0ire commented:
@jderusse I fixed the CS with `vendor/bin/phpcbf`, then addressed the static analysis issues that were about unnecessary casts and checks, now we know for sure we have an array, and added the unit test that was missing in #8201

jderusse commented:
> @jderusse I fixed the CS with `vendor/bin/phpcbf`, then addressed the static analysis issues that were about unnecessary casts and checks, now we know for sure we have an array, and added the unit test that was missing in #8201 .

ohh.. thank you so much

SenseException reviewed (approved):
Returning an exception instance is a bug, so I think this should be fine for 2.9.x.

oojacoboo commented:
@greg0ire yikes - don't recall that exactly, but good call on returning a single type.
